sequencer/src/block/namespace_payload/ns_payload/test.rs

```diff
@@ -0,0 +1,290 @@
+    }
+
+    // extremely small payload can't even fit the tx table header
+    {
+        for i in 0..NsPayloadBuilder::tx_table_header_byte_len() {
+            let ns_payload = NsPayloadOwned::entries_total(&[10, 20, 30], i);
+            assert_eq!(ns_payload.iter().count(), 0);
+        }
+    }
+}
+
+#[test]
+fn negative_len_txs() {
+    setup_logging();
+    setup_backtrace();
+    let ns_id = NamespaceId::from(69); // dummy
+
+    // 1 negative-length tx at the end, no overlapping tx bytes
+    {
+        let ns_payload = NsPayloadOwned::entries_body(&[10, 30, 20], 30);
+        let txs = ns_payload.export_all_txs(&ns_id);
+        assert_eq!(txs.len(), 3);
+        assert_eq!(txs[0].payload(), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
+        assert_eq!(
+            txs[1].payload(),
+            [10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29]
+        );
+        assert!(txs[2].payload().is_empty());
+    }
+
+    // 1 negative-length tx in the middle, overlapping tx bytes
+    {
+        let ns_payload = NsPayloadOwned::entries_body(&[20, 10, 30], 30);
+        let txs = ns_payload.export_all_txs(&ns_id);
+        assert_eq!(txs.len(), 3);
+        assert_eq!(
+            txs[0].payload(),
+            [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
+        );
+        assert!(txs[1].payload().is_empty());
+        assert_eq!(
+            txs[2].payload(),
+            [10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29]
+        );
+    }
+
+    // 1 negative-length tx in the middle, one tx contains all others
+    {
+        let ns_payload = NsPayloadOwned::entries_body(&[30, 10, 20], 30);
+        let txs = ns_payload.export_all_txs(&ns_id);
+        assert_eq!(txs.len(), 3);
+        assert_eq!(
+            txs[0].payload(),
+            [
+                0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
+                23, 24, 25, 26, 27, 28, 29
+            ]
+        );
+        assert!(txs[1].payload().is_empty());
+        assert_eq!(txs[2].payload(), [10, 11, 12, 13, 14, 15, 16, 17, 18, 19]);
+    }
+
+    // all txs negative-length except the first
+    {
+        let ns_payload = NsPayloadOwned::entries_body(&[30, 20, 10], 30);
+        let txs = ns_payload.export_all_txs(&ns_id);
+        assert_eq!(txs.len(), 3);
+        assert_eq!(
+            txs[0].payload(),
+            [
+                0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
+                23, 24, 25, 26, 27, 28, 29
+            ]
+        );
+        assert!(txs[1].payload().is_empty());
+        assert!(txs[2].payload().is_empty());
+    }
+}
+
+#[test]
+fn tx_table_header() {
+    setup_logging();
+    setup_backtrace();
+    let ns_id = NamespaceId::from(69); // dummy
+
+    // header declares 1 fewer txs, tx table bytes appear in tx payloads, wasted
+    // pßayload bytes
```

Comment:
```suggestion
    // payload bytes
```

Reply:
I'll fix this in a future PR so as not to trigger CI